inspirehep/dojson/hepnames/fields/bd1xx.py

```diff
@@ -126,9 +131,27 @@ def name2marc(self, key, value):
 @utils.for_each_value
 def ids(self, key, value):
     """All identifiers, both internal and external."""
+    IDS_MAP = {
+        'ARXIV': 'ARXIV',
+        'BAI': 'INSPIRE BAI',
+        'CERN': 'CERN',
+        'DESY': 'DESY',
+        'INSPIRE': 'INSPIRE ID',
+        'KAKEN': 'KAKEN',
+        'ORCID': 'ORCID',
+        'SLAC': 'SLAC',
+        'WIKIPEDIA': 'WIKIPEDIA',
+    }
+
+    type_ = IDS_MAP.get(value.get('9').upper())
```

Comment:
There are (potentially) two problems here:

1. Multiple `035__9`
2. No `035__9`.

Your solution is guarding only against the second. The fact is, I can't find examples of either, so I guess that we can do it unsafely and collect the errors on Sentry.